java/core/src/test/java/com/lancedb/lance/TestUtils.java

```diff
@@ -127,14 +129,18 @@ public void createDatasetAndValidate() throws IOException, URISyntaxException {
           assertEquals(ROW_COUNT, dataset.countRows());
           schema = reader.getVectorSchemaRoot().getSchema();
           validateFragments(dataset);
+          assertEquals(1, dataset.version());
+          assertEquals(1, dataset.latestVersion());
```

Comment:
Do we have tests where `version() != latestVersion()`?

Reply:
Addressed the comment, PTAL